src/core/dev_api/openvino/runtime/shared_buffer.hpp

```diff
@@ -28,4 +28,51 @@ class SharedBuffer : public ov::AlignedBuffer {
     T _shared_object;
 };
 
+/// \brief SharedStreamBuffer class to store pointer to pre-acclocated buffer and provide streambuf interface.
+class SharedStreamBuffer : public std::streambuf {
+public:
+    SharedStreamBuffer(char* data, size_t size) : m_data(data), m_size(size), m_offset(0) {}
```

Comment:
'data' is not passed to std::streambuf, so operator>> for std::istream should not work in [core_impl](https://github.com/openvinotoolkit/openvino/pull/26696/files#diff-8d13c3a18a5f54a062387b01b0e2bb6dedc35c42e73c752ae00b73ee4c60f0fbR1419) and HeaderException should be thrown. Actually, it could be done on Linux only, due to the std::streambuf has different implementations depend on OS. 
Another way is to define operator>> but you should handle end of line for different OS at least.

Reply:
As I can see operator>>  works well on both Windows and Linux. It works via overrided underflow() and uflow() calls

Reply:
Can we rely on a third-party internal implementation? It may be changed in future releases.